enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -2822,6 +2825,442 @@ class AdjointGenerator
                        << I;
         report_fatal_error("(reverse) unknown intrinsic");
       }
+      return;
+    }
+    case DerivativeMode::ForwardMode: {
+
+      IRBuilder<> Builder2(&I);
+      getForwardBuilder(Builder2);
+
+      switch (ID) {
+      case Intrinsic::nvvm_barrier0_popc:
+      case Intrinsic::nvvm_barrier0_and:
+      case Intrinsic::nvvm_barrier0_or:
+      case Intrinsic::nvvm_barrier0:
+      case Intrinsic::amdgcn_s_barrier:
+      case Intrinsic::nvvm_membar_cta:
+      case Intrinsic::nvvm_membar_gl:
+      case Intrinsic::nvvm_membar_sys:
+      case Intrinsic::assume:
+      case Intrinsic::prefetch:
+      case Intrinsic::dbg_declare:
+      case Intrinsic::dbg_value:
+#if LLVM_VERSION_MAJOR > 6
+      case Intrinsic::dbg_label:
+#endif
+      case Intrinsic::dbg_addr:
```

Comment:
I think this would be better handled by a check for inactive value and instruction rather than the cases here.